qrenderdoc/Windows/EventBrowser.cpp

```diff
@@ -5379,6 +5394,52 @@ void EventBrowser::highlightBookmarks()
   }
 }
 
+void EventBrowser::bookmarkContextMenu(QRClickToolButton *button, uint32_t EID)
+{
+  QMenu contextMenu(this);
+
+  QAction renameBookmark(tr("&Rename"), this);
+  QAction deleteBookmark(tr("&Delete"), this);
+
+  renameBookmark.setIcon(Icons::page_white_edit());
+  deleteBookmark.setIcon(Icons::del());
+
+  contextMenu.addAction(&renameBookmark);
+  contextMenu.addAction(&deleteBookmark);
+
+  QObject::connect(&deleteBookmark, &QAction::triggered, [this, EID]() {
+    m_Ctx.RemoveBookmark(EID);
+    repopulateBookmarks();
+  });
+
+  QObject::connect(&renameBookmark, &QAction::triggered, [this, button, EID]() {
+    EventBookmark editedBookmark;
+    for(const EventBookmark &bookmark : m_Ctx.GetBookmarks())
+    {
+      if(bookmark.eventId == EID)
+      {
+        editedBookmark = bookmark;
+        break;
+      }
+    }
+
+    if(editedBookmark.eventId == EID)
+    {
+      bool ok;
+      editedBookmark.text = QInputDialog::getText(
+          this, lit("Rename"), lit("New name:"), QLineEdit::Normal,
+          editedBookmark.text.empty() ? QString::number(EID) : (QString)editedBookmark.text, &ok);
+      if(ok && !editedBookmark.text.empty())
```

Comment:
Minor point, should we allow setting the text to empty to revert back to the plain-EID named bookmark? i.e removing the text but keeping the bookmark?

Reply:
That's a good idea, I'll change it.